daemon/volumes.go

```diff
@@ -245,6 +248,34 @@ func (daemon *Daemon) registerMountPoints(container *container.Container, hostCo
 			}
 		}
 
+		if mp.Type == mounttypes.TypeImage {
+			if !daemon.Config().CommonConfig.Features["image-mount"] {
+				return fmt.Errorf("Feature 'image-mount' is not enabled")
+			}
+			img, err := daemon.imageService.GetImage(ctx, mp.Source, backend.GetImageOpts{})
+			if err != nil {
+				return err
+			}
+
+			rwLayerOpts := &layer.CreateRWLayerOpts{
+				StorageOpt: container.HostConfig.StorageOpt,
+			}
+
+			layerName := fmt.Sprintf("%s-%s", container.ID, mp.Source)
+			layer, err := daemon.imageService.CreateLayerFromImage(img, layerName, rwLayerOpts, nil)
+			if err != nil {
+				return err
+			}
+			path, err := layer.Mount("")
+			if err != nil {
+				return err
+			}
+
+			mp.Name = mp.Spec.Source
+			mp.Spec.Source = img.ID().String()
+			mp.Source = path
```

Comment:
```suggestion
			mp.Source = path
			mp.RW = false
```

Let's make it read-only explicitly